app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/PreviewGenerators.kt

```diff
@@ -59,6 +75,33 @@ object PreviewGenerators {
             PREVIEWS,
             "No generators found for type .${extension(path)} ($path)"
         )
+        // This section called when the code will failed to generate preview for
+        // pdf, link, txt extension.This code will match with existing factories'
+        // acceptedExtensions as well as acceptedMimeTypes (if ext is blank).
+        if (
+            ext.isBlank() && getMimeTypeUsingTika(path = path) == "application/pdf"
+        ) {
+            generatorsByExt["pdf"]?.let { generator ->
```

Comment:
Looks like code duplication again, code duplication should be avoided at all cost, because it's difficult to maintain (easy to modify one place and forget about another). Maybe we need to create additional map `generatorByMIME`, fill it with the same values of `generatorByExt` just for different keys, check one map, then another, then invoke necessary code one time instead of several.

Also, no generator should be retrieved directly by mentioning String constant, like `"pdf"`. The maps should be used with variables, otherwise we can just store pdf generator into class constant and use it instead of searching in the map.

Reply:
Ok. I will change and call the method direct as it will more useful.